demo/transcribe_app.py

```diff
@@ -1,31 +1,86 @@
+    # "kostissz/whisper-tiny-el",  # custom HF model example
+    # "artifacts/whisper-small-en",  # local model directory example
 ]
 
 
-def load_model(model_id: str, language: str) -> Tuple[Pipeline, str]:
-    if model_id and language:
-        yield None, f"Loading {model_id}..."
-        pipe = pipeline(
-            "automatic-speech-recognition",
-            model=model_id,
-            generate_kwargs={"language": language},
+def _load_local_model(model_dir: str, language: str) -> Tuple[Pipeline | None, str]:
+    if not Path(model_dir).is_dir():
+        return None, f"⚠️ Couldn't find local model directory: {model_dir}"
+    from transformers import (
+        WhisperProcessor,
+        WhisperTokenizer,
+        WhisperFeatureExtractor,
+        WhisperForConditionalGeneration,
+    )
+
+    processor = WhisperProcessor.from_pretrained(model_dir)
+    tokenizer = WhisperTokenizer.from_pretrained(
+        model_dir, language=language, task="transcribe"
+    )
+    feature_extractor = WhisperFeatureExtractor.from_pretrained(model_dir)
+    model = WhisperForConditionalGeneration.from_pretrained(model_dir)
+
+    return pipeline(
+        task="automatic-speech-recognition",
+        model=model,
+        processor=processor,
+        tokenizer=tokenizer,
+        feature_extractor=feature_extractor,
+    ), f"✅ Local model has been loaded from {model_dir}."
+
+
+def _load_hf_model(model_repo_id: str, language: str) -> Tuple[Pipeline | None, str]:
+    if not hf_model_exists(model_repo_id):
+        return (
+            None,
+            f"⚠️ Couldn't find {model_repo_id} on Hugging Face. If its a private repo, make sure you are logged in locally.",
         )
-        yield pipe, f"✅ Model {model_id} has been loaded."
+    return pipeline(
+        "automatic-speech-recognition",
+        model=model_repo_id,
+        generate_kwargs={"language": language},
+    ), f"✅ HF Model {model_repo_id} has been loaded."
+
+
+def load_model(
+    language: str, dropdown_model_id: str, hf_model_id: str, local_model_id: str
+) -> Tuple[Pipeline, str]:
+    if dropdown_model_id and not hf_model_id and not local_model_id:
+        yield None, f"Loading {dropdown_model_id}..."
+        yield _load_hf_model(dropdown_model_id, language)
+    elif hf_model_id and not local_model_id and not dropdown_model_id:
+        yield None, f"Loading {hf_model_id}..."
+        yield _load_hf_model(hf_model_id, language)
+    elif local_model_id and not hf_model_id and not dropdown_model_id:
+        yield None, f"Loading {local_model_id}..."
+        yield _load_local_model(local_model_id, language)
     else:
-        yield None, "⚠️ Please select a model and a language from the dropdown"
+        yield (
+            None,
+            "️️⚠️ Please select or fill at least and only one of the three options above",
+        )
+    if not language:
+        yield None, "⚠️ Please select a language from the dropdown"
```

Comment:
I think it would be fine to not require this (would simplify the UI). People will most likely try a model finetuned for a specific language and the autodetect of whisper works just fine

Reply:
Actually the issue is that whisper also does translation, so there are cases like [this](https://discuss.huggingface.co/t/how-to-set-audio-language-in-whisper-pipeline/25914), where the audio might be in one language, but the text output will be forced to English, so I think setting the language is helpful, if not necessary.